### ISSUE TITLE
Clarify the coding standard about naming

### ISSUE THREAD

**Issue by VincentLanglet** [twigphp/Twig#3545]
TLDR: I think the twig coding standard about variable casing should be either removed, either precised.

In the twig coding standard https://github.com/twigphp/Twig/blob/3.x/doc/coding_standards.rst#coding-standards, there is the following rule 

> Use lower cased and underscored variable names.
```
{% set foo = 'foo' %}
{% set foo_bar = 'foo' %}
```

But there is nothing about
- Variable passed by the controller to the view
- Method/filter names
- Getter usage

Since twig started to take a decision about the right casing in twig files, IMHO it should be more precise about all the possible situation ; or in the contrary, it shouldn't start to take a decision.

### Variable passed by the controller to the view

Currently in controller I only saw camelCase variable passed to the view, like https://github.com/symfony/demo/blob/main/src/Controller/BlogController.php#L62
Which means we would end up with both camelCased and snakeCased variable in the same template.

I'm not sure it's the best.
If I used a reusable template which need some value `FooBar`, should I then call is `foo_bar` or instead `fooBar` ?
Let's say `foo_bar` is recommended then if I use this template, I could have to add `with {foo_bar: fooBar}` if I use it with a variable passed by the controller...

So should snake_case` be recommended for the value passed from the controller to the view ?

### Method/filter names

Looking at all the method/filter provided by twig, like `html_to_markdown` it seems like snakeCase is use.
I think it could be added to the coding standard

### Getter usage

Let's say I have a class Foo with a method getMyBar.
Should I write `foo.myBar` or `foo.mybar` or `foo.my_bar` ? (I don't even know if the last one is supported)

I currently see a lot camelCase for this situation. Let's say the class is `SuperFoo` now.
I could end with code like this
```
{% set super_foo = ... %}
{% set my_bar = super_foo.myBar %}
```
So it leads to using both camelCase and snakeCase in the same expression. Not sure is great.

**Comment by VincentLanglet** [comment]
Friendly ping @fabpot, would it be possible to have more information about the coding standard of twig about naming ?
I'll be happy to provide a PR to improve the doc then.

**Comment by filecage** [comment]
I'm adding a note here to an issue I've created one year ago which addresses a similar thing: the internal "normalisation" of argument casing in #3475. This is beyond clarification in standards, but also adresses possible confusion and is a push towards "remove this standard".

**Comment by leevigraham** [comment]
Symonfy "Best Practices" https://symfony.com/doc/current/best_practices.html#templates

### [Use Snake Case for Template Names and Variables](https://symfony.com/doc/current/best_practices.html#use-snake-case-for-template-names-and-variables)

Use lowercase snake_case for template names, directories, and variables (e.g. `user_profile` instead of `userProfile` and `product/edit_form.html.twi`g instead of `Product/EditForm.html.twig`).

### [Prefix Template Fragments with an Underscore](https://symfony.com/doc/current/best_practices.html#prefix-template-fragments-with-an-underscore)

Template fragments, also called "partial templates", allow to [reuse template contents](https://symfony.com/doc/current/templates.html#templates-reuse-contents). Prefix their names with an underscore to better differentiate them from complete templates (e.g. `_user_metadata.html.twig` or `_caution_message.html.twig`).

**Comment by VincentLanglet** [comment]
> Symonfy "Best Practices" [symfony.com/doc/current/best_practices.html#templates](https://symfony.com/doc/current/best_practices.html#templates)
> 
> ### [Use Snake Case for Template Names and Variables](https://symfony.com/doc/current/best_practices.html#use-snake-case-for-template-names-and-variables)
> Use lowercase snake_case for template names, directories, and variables (e.g. `user_profile` instead of `userProfile` and `product/edit_form.html.twi`g instead of `Product/EditForm.html.twig`).

Hi @leevigraham, I'm fully aware about this standard.

I was pointing out the inconsistency between
- `{% set foo_bar = 42 %}` => Here Symfony recommends lowercase.
- Variables coming from controllers => Here Symfony use camelCase, example https://github.com/symfony/demo/blob/a53f9304bd9b1ded5a44a3de51263a7de62a50d4/src/Controller/BlogController.php#L67
- `{{ foo_bar.getSomething() }}` => since getter are written in PHP and are using camelCase, we're mixing snake_case and camelCase in the same expression.

That feels odd.

**Comment by fabpot** [comment]
> So should snake_case` be recommended for the value passed from the controller to the view ?

Yes, all variable names should be using snake_case.

> Looking at all the method/filter provided by twig, like html_to_markdown it seems like snakeCase is use.

Yes, filter/function/test names should be using snake_case.

For getters, it's a bit more complicated as methods in PHP are not using snake_case.

**Comment by fabpot** [comment]
See #4314
